spec/mondrian_spec.rb

```diff
@@ -188,6 +188,29 @@ def restore_property_values
     result.formatted_values.first.should == Date.parse(date_value.to_s).strftime("%d.%m.%Y")
   end
 
+  describe "#null member" do
+    before(:all) do
+      set_property_values(
+        "mondrian.rolap.ignoreInvalidMembers" => "true",
+        "mondrian.rolap.ignoreInvalidMembersDuringQuery" => "true"
+      )
+    end
+
+    after(:all) do
+      restore_property_values
+    end
+
+    it "should return no results if slice axis has #null member" do
+      result = @olap.from('Sales').
+      columns('[Measures].[Unit Sales]').
+      rows('[Customers].[Name].Members').
+      where('[Time].[3000]').execute
```

Comment:
I'm not sure that this should return no columns if we explicitly select them. If we would add a row filter by this measure that it should not return no rows.

Reply:
Also I'm not sure that the Java code is correct
```java
for ( Exp expr : ((ResolvedFunCall) slicerAxisExp).getArgs() ) {
  if (expr instanceof MemberExpr) {
    MemberExpr memberExpr = (MemberExpr) expr;
    Member member = memberExpr.getMember();
    return member.isNull();
  }
}
```
I suspect that we need to check that if member.isNull() is false then we need to check other expr arguments.

Also we need to add more tests e.g. with several dimensions in `where('[Customers].[USA]', '[Time].[3000]')` or using `where('[Customers].[USA]').crossjoin('[Time].[2011]', '[Time].[3000]')` (which should return results).